examples/custom-image-dataset/src/downloader.rs

```diff
@@ -0,0 +1,46 @@
+use indicatif::{ProgressBar, ProgressState, ProgressStyle};
+use reqwest::Client;
+use std::cmp::min;
+use std::io::Write;
+
+/// Download the file at the specified url to a bytes vector.
+/// File download progress is reported with the help of a [progress bar](indicatif).
+///
+/// Taken from [burn-dataset](https://github.com/tracel-ai/burn/blob/main/burn-dataset/src/vision/downloader.rs).
+#[tokio::main(flavor = "current_thread")]
+pub async fn download_file_as_bytes(url: &str, message: &str) -> Vec<u8> {
```

Comment:
It seems to be duplicated from the `mnist` dataset implementation right? Maybe we can actually add cifar to `burn-dataset` and reuse the setup from mnist?

Reply:
Well the code is technically not duplicated since both call that function which is separated in its own module :) but yes, both use the same function (intentionally) and that download function can be used for any other future dataset we might want to add.

I wasn't sure if we wanted to add CIFAR-10 explicitly so I only put it as an example, but it could be refactored. And actually the original CIFAR-10 source is in bytes form (I used a mirror with the folder structure to illustrate the `ImageFolderDataset` which is more common).